jerry-ext/context/include/jerryscript-ext/context.h

```diff
@@ -0,0 +1,55 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_CONTEXT_H
+#define JERRYX_CONTEXT_H
+
+#include "jerryscript.h"
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its second parameter
+ */
+void *jerryx_context_init (void);
+
+/**
+ * Function to pass to jerry_init_with_user_context() as its third parameter
+ */
+void jerryx_context_deinit (void *user_context_p);
+
+typedef struct
+{
+  /** the initialization function */
+  jerry_user_context_init_cb init_cb;
+  /** the deinitialization function */
+  jerry_user_context_deinit_cb deinit_cb;
+  /** the index of the slot */
+  int index;
```

Comment:
I've tried that at https://gist.github.com/gabrielschulhof/2dfe8386a4a4924e9582bba1cb7f3123#file-section-picture-c-L23

and I get the error
```
section-picture.c:23:22: error: initializer element is not computable at load time
 static int x_index = (&x - __start_struct_array);
```